### DISCUSSION TITLE
Modular Deployments for BICEP

### DISCUSSION THREAD

**Discussion by FallenHoot** [Azure/avdaccelerator#348] [Ideas]
Having modular deployments will allow for loosely coupled deployments, accommodate variations in configuration, and shorter product life cycles.

When it comes to using IaC and creating pipelines, you can split the monolithic deployment (deploy-baseline.bicep) into several modular deployments. It allows for readability and customization without deep diving into 1,400+ lines of code.

Having deployment groups such as:
zone-0 - foundation (Management Group, Subscription, Resource Groups, Networking, AVD Management Plane/Host pool)
zone-1 - AVD Session Host()
zone-2 - AVD  Application Groups
zone-3 - AVD Workspaces
zone-4 - Scaling Plans
zone-5 - FSlogix (FSLogix, Identity)
zone-6 - MSIX

Doing it this way makes the code understandable without going through 100 parameters of things you don't need.

**Comment by danycontre** [comment]
@FallenHoot thanks for the feedback.

Agree with your comments, we have been working in modularizing the code for Bicep, but we still have a long way to go. 

ARM template will always be a single template because we work only on the Bicep code and build ARM from it.

PS: terraform code already introduced modules.

cc: @swathibhat1

**Reply by FallenHoot** [reply]
I am working on something now around this and happy to share it with you once it is completed.
